### PR TITLE
Change Search Operator

### PR DESCRIPTION
**Pull request by RFSH**
Previously we were using actual `::ciregex::` operator for search, the problem with that usage was that we were manipulating the searchTerm internally so ermrest understands it. For instance when we search for `11`, what we need to send to ermrest is `^(.*[^0-9.])?0*11([^0-9].*|$)` regular expression. So we needed to keep two versions of filter, one that users see and one that ermrest understand.

The uri that parser returns (`Location.uri`) is what the client (chaise) understand and other apis for ermrest uri (`Location.ermrsetUri`, `Location.ermrestPath`, etc.) are introduced.

Issue: #455

### CODE REVIEW

File: js/parser.js [hunk 1]
```diff
@@ -145,52 +146,45 @@ var ERMrest = (function(module) {
         }
 
         // Split compact path on '/'
-        // Expected format: "[schema_name:]table_name[/{attribute::op::value}{&attribute::op::value}*]"
+        // Expected format: "<schema:table>/<filter(s)>/<joins(s)/<search>"
```
Comment (jrchudy) [inline js/parser.js]: Not that this matters much but you forgot a `>` after `join(s)`

File: js/parser.js [hunk 2]
```diff
@@ -283,25 +277,144 @@ var ERMrest = (function(module) {
...
         toString: function(){
-            return this._uri;
+            return this.uri;
         },
 
 
         /**
-         *
+         * <service>/catalog/<catalogId>/<api>/<projectionSchema:projectionTable>/<filters>/<joins>/<search>/<sort>/<page>/<queryParams>
          * @returns {String} The full URI of the location
```
Comment (jrchudy) [inline js/parser.js]: How does this URI differ from `ermrestUri`? Could you mention that in the comments here.

File: js/parser.js [hunk 3]
```diff
@@ -283,25 +277,144 @@ var ERMrest = (function(module) {
...
+            if (this._uri === undefined) {
+                this._uri = this.compactUri + this._modifiers;
+            }
             return this._uri;
         },
 
         /**
-         *
+         * <service>/catalog/<catalogId>/<api>/<projectionSchema:projectionTable>/<filters>/<joins>/<search>
          * @returns {String} The URI without modifiers or queries
```
Comment (jrchudy) [inline js/parser.js]: Same comment here. How does this compact URI differ from `ermrestCompactUri`?

**Comment by RFSH** [comment]
@jrchudy Added some comments to the apis. The only difference between `ermrestUri` and `uri` is the search operator right now. `uri` might have `*::search::` in it, but `ermrestUri` shouldn't. However, they will be more and more different by introducing facets.

[There is a test case](https://github.com/informatics-isi-edu/ermrestjs/blob/master/test/specs/reference/tests/13.search.js#L412) for that in ermrestjs and which expects the existing behavior (What recordset is showing in the textbox is `searchTerm` and this expects the `searchTerm` to include the regular expression). We can add some test cases for it in chaise too. We don't need to go to next page and get back, we can just click on `Permalink` and then check the search box and url.

**Pull request merged**